### PR TITLE
Skeletons to load Macauff parameters

### PR DESCRIPTION
**Pull request by camposandro**
Adds the class skeletons to load the Macauff configuration parameters (both all sky and catalog-specific). Also creates the stubs for the algorithm class that LSDB will be able to invoke to perform the crossmatch.

### CODE REVIEW

**Review by delucchi-cmu — DISMISSED** [review]

File: tests/data/all_sky_params.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+pos_corr_dist: 11
+# Convolution (fourier transform) parameters
+# Integer number of real space grid points, for Hankel transformations
+real_hankel_points: 10000
+# Integer number of fourier space grid points
+four_hankel_points: 10000
+# Maximum fourier space "rho" parameter considered (typically larger than the inverse of the smallest Gaussian sigma)
+four_max_rho: 100
+# Maximum extent of cross-match, used in non-all-sky cases to remove sources suffering potential edge effects -- min/max first axis coordinates (ra/l) then min/max second axis coordinates (dec/b)
+cross_match_extent:
```
Comment (Onoddil) [inline tests/data/all_sky_params.yml:19]: `cross_match_extent` is a per-pixel parameter and should be moved to those yml files if/when we include them.

**Review by Onoddil — CHANGES_REQUESTED** [review]
Apart from the one unnecessary all-sky parameter this looks good!

File: src/lsdb_macauff/config/all_sky_params.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+@dataclass
+class AllSkyParams:
+    """Joint sky parameters"""
+
+    include_perturb_auf: bool
+    include_phot_like: bool
+    use_phot_priors: bool
+    cross_match_extent: List[float]
```
Comment (Onoddil) [inline src/lsdb_macauff/config/all_sky_params.py:14]: Remove `cross_match_extent` here as well as in the `yml` file

Comment (camposandro) [reply src/lsdb_macauff/config/all_sky_params.py]: Thanks @Onoddil, just did!

**Review by Onoddil — APPROVED** [review]

**Pull request merged**